File: app/src/main/java/com/example/project_note/DataBase/NoteDatabase.java
package com.example.project_note.DataBase;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Note.class}, version = 1)
public abstract class NoteDatabase extends RoomDatabase {

    private static final  String DATABASE_NAME = "user.db";
    private static NoteDatabase instance;

    public static synchronized NoteDatabase getInstance(Context context){
        if(instance == null){
            instance = Room.databaseBuilder(context.getApplicationContext(), NoteDatabase.class,DATABASE_NAME)
                    .allowMainThreadQueries()
                    .build();
        }
        return instance;
    }
    public abstract NoteDAO noteDAO();
}

File: app/src/main/java/com/example/project_note/DataBase/NoteDatabase.kt
package com.example.project_note.DataBase

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = [Note::class, Image::class], version = 1)
//@TypeConverters(Converters::class)

abstract class NoteDatabase : RoomDatabase() {
    abstract fun noteDAO(): NoteDAO
    abstract fun ImageDAO(): ImageDAO


    companion object {
        private val DATABASE_NAME = "user.db"
        private var instance: NoteDatabase? = null

        @Synchronized
        fun getInstance(context: Context): NoteDatabase {
            if (instance == null) {
                instance = databaseBuilder(
                    context.applicationContext,
                    NoteDatabase::class.java,
                    DATABASE_NAME
                )
                    .allowMainThreadQueries()
                    .build()
            }
            return instance!!
        }
    }
}